common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/WorkerSupervisor.cs

```diff
@@ -102,47 +104,53 @@ public class WorkerSupervisor : IWorkerSupervisor, IDisposable {
         /// </summary>
         /// <returns>awaitable task</returns>
         private async Task StopWorker() {
-            // sort workers, so that a worker in state Stopped, Stopping or WaitingForJob will terminate first 
+            // sort workers, so that a worker in state Stopped, Stopping or WaitingForJob will terminate first
             var worker = _instances.OrderBy(kvp => kvp.Key.Status).First();
             var workerId = worker.Key.WorkerId;
             _logger.Information("Stopping worker with id {WorkerId}", workerId);
             _instances.TryRemove(worker.Key, out _);
-            await worker.Key.StopAsync();
+            await worker.Key?.StopAsync();
             worker.Value?.Dispose();
         }
 
         /// <inheritdoc/>
         public void Dispose() {
             if (_ensureWorkerRunningTimer != null) {
-                _ensureWorkerRunningTimer.Stop();
+                _ensureWorkerRunningTimer.Enabled = false;
```

Comment:
Why is there no need to call `_ensureWorkerRunningTimer.Stop()` now ?

Reply:
Ah, because `Stop()` just sets `Enabled` to `false`.

Reply:
_ensureWorkerRunningTimer.Stop() is actually the same as: _ensureWorkerRunningTimer.Enabled = false. I made this change consistently through this class